src/backend/gpopt/utils/COptTasks.cpp

```diff
@@ -498,7 +498,9 @@ COptTasks::OptimizeTask(void *ptr)
 		CMDCache::Init();
 		CMDCache::SetCacheQuota(optimizer_mdcache_size * 1024L);
 	}
-	else if (reset_mdcache)
+	else if (reset_mdcache ||
+			 (gpdb::GPDBTransactionIdIsValid(CMDCache::GetTransientXmin()) &&
```

Comment:
Could we remove this line?

Reply:
what do you mean?

Reply:
such condition only
```c
else if (reset_mdcache || gpdb::GetTransactionXmin() != CMDCache::GetTransientXmin())
```

Reply:
then if the current transaction has not set `TransientXmin` for the cache, then we will reset the cache on each query. therefore, no.